ethereum/test/bridge.js

```diff
@@ -678,11 +820,200 @@ contract("Bridge", function () {
         });
     })
 
-    it("should burn bridged assets wrappers on transfer to another chain", async function () {
+    it("should handle additional data on token bridge transfer with payload in single transaction when feeRecipient == transferRecipient", async function () {
+        const accounts = await web3.eth.getAccounts();
+        const amount = "1000000000000000000";
+        const fee = "1000000000000000";
+        const feeRecipient = accounts[0]; // same account as sender to check feeRecipient != transferRecipient condition
+
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+
+        const wrappedAddress = await initialized.methods.wrappedAsset("0x" + testBridgedAssetChain, "0x" + testBridgedAssetAddress).call();
+        const wrappedAsset = new web3.eth.Contract(TokenImplementation.abi, wrappedAddress);
+
+        const accountBalanceBefore = await wrappedAsset.methods.balanceOf(accounts[0]).call();
+        const totalSupplyBefore = await wrappedAsset.methods.totalSupply().call();
+
+        // we are using the asset where we created a wrapper in the previous test
+        const data = "0x" +
+            "03" +
+            // amount
+            web3.eth.abi.encodeParameter("uint256", new BigNumber(amount).div(1e10).toString()).substring(2) +
+            // tokenaddress
+            testBridgedAssetAddress +
+            // tokenchain
+            testBridgedAssetChain +
+            // receiver (must be self msg.sender)
+            web3.eth.abi.encodeParameter("address", accounts[0]).substr(2) +
+            // receiving chain
+            web3.eth.abi.encodeParameter("uint16", testChainId).substring(2 + (64 - 4)) +
+            // fee
+            web3.eth.abi.encodeParameter("uint256", new BigNumber(fee).div(1e10).toString()).substring(2) +
+            // additional payload
+            web3.eth.abi.encodeParameter("address", accounts[1]).substr(2);
```

Comment:
In the above examples, we have been using arbitrary payloads, is there a reason why we're using address in this context?

Reply:
There's a few examples below, but mentioning here just so I can understand.

Reply:
no particular reason here, just matches what we do with the mock integration which actually uses that address to send the funds to